Assets/Scripts/StateManager.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using UnityEngine;
+
+namespace Laz
+{
+    public enum State
+    {
+        Pause,
+        Play,
+        Death,
+        EndGame
+    }
+
+    public interface IStateManager
+    {
+        event Action<State> OnStateChanged;
+        State GetState();
+
+        void SetState(State state);
+    }
+
+    /// <summary>
+    /// The 'Singleton' class
+    /// </summary>
+    public class StateManager
```

Comment:
Should this be inheriting from IStateManager? If not, where are we using IStateManager?

Reply:
oh yeah it should be.

Reply:
Updated